libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java

```diff
@@ -367,8 +358,50 @@ private boolean hasNetworkConnection() {
     if (connectivityManager == null) {
       return false;
     }
-
     NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo();
     return activeNetwork != null && activeNetwork.isConnectedOrConnecting();
   }
+
+  private void sendEventFeedback(FeedbackItem feedbackItem) {
+    if (navigationViewEventDispatcher != null) {
+      navigationViewEventDispatcher.onFeedbackSent(feedbackItem);
+    }
+  }
+
+  private void sendEventArrival(RouteProgress routeProgress, Milestone milestone) {
+    if (navigationViewEventDispatcher != null && RouteUtils.isArrivalEvent(routeProgress, milestone)) {
+      navigationViewEventDispatcher.onArrival();
+    }
+  }
+
+  private void sendEventOffRoute(Point newOrigin) {
+    if (navigationViewEventDispatcher != null && navigationViewEventDispatcher.allowRerouteFrom(newOrigin)) {
+      navigationViewEventDispatcher.onOffRoute(newOrigin);
+      OffRouteEvent event = new OffRouteEvent(newOrigin, routeProgress);
+      navigationViewRouteEngine.fetchRouteFromOffRouteEvent(event);
+      isOffRoute.setValue(true);
+    }
+  }
+
+  private void sendEventIsRunning(boolean isRunning) {
```

Comment:
Should this be `sendEventIfRunning`? It's kind of a confusing name, maybe `sendStatusEvent`?